lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -425,6 +439,32 @@ class ReportUIFeatures {
         this._toggleDarkTheme();
         break;
       }
+      case 'swap-locale': {
+        if (!this._swapLocaleOptions) throw new Error('must call .initSwapLocale first');
+
+        // TODO: esmodules would be nice here...
+        // @ts-ignore
+        if (!window.Lighthouse || !window.Lighthouse.i18n) await import('./i18n-module.js');
+
+        // Waiting for UI :)
+        const randomLocales = ['es', 'fr', 'vi', 'en-US'];
+        window.__locale_index = window.__locale_index || 0;
+        const locale = randomLocales[window.__locale_index++ % randomLocales.length];
+
+        // TODO: locales.js maps a locale code to a locale module. Need that same mapping,
+        // but for locale code to locale module _name_, so that we can fetch w/ that same
+        // name here.
+        const localModuleName = locale;
+        const lhlMessages = await this._swapLocaleOptions.fetchData(localModuleName);
```

Comment:
could skip this fetch for subsequent requests for the same locale, but would need a method to query if a property in `LOCALES` exists. not an important optimization so I think better to ignore it.

Reply:
could also just create a cache in the closure of this fn we invoke 🤷‍♂